tests/Paramore.Brighter.Core.Tests/CommandProcessors/TestDoubles/SpyOutbox.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Paramore.Brighter.Observability;
+
+namespace Paramore.Brighter.Core.Tests.CommandProcessors.TestDoubles
+{
+    internal class SpyOutbox : 
```

Comment:
Could we use the InMemoryOutbox? We are trying to avoid Test Doubles where the in-memory version could substitute. If we need to query the state of the outbox, is that something that is all outboxed,  or should the in-memory outbox be exposed?

Reply:
The issue with using the `InMemoryOutbox` is that it's the default implementation, and the scenario being tested here is when the user uses something _other_ than the default, since that was what led to this issue being found in the first place (I also wouldn't have found that we should be clearing the bound `MethodInfo`'s when we clear the outbox mediator from the `CommandProcessor`). 

To do that without a test-only `Spy` implementation we would need to either have an additional type of in memory outbox in the core package (which feels redundant?), or pull in one of the other outbox implementations (and then require all the associated infrastructure in order to be able to run the core tests). This feels like the cleanest solution to me.

Reply:
Ah yes, that makes sense in this context. Closed.